content/memory_model.tex

```diff
@@ -37,11 +35,41 @@
 Objects} on a special memory region called the \emph{Symmetric Heap}. The
 Symmetric Heap is created during the execution of a program at a memory location
 determined by the implementation. The Symmetric Heap may reside in different
-memory regions on different \acp{PE}. Figure~\ref{fig:mem_model} shows how
-\openshmem implements a \ac{PGAS} model using remotely accessible symmetric
-objects and private data objects when executing an \openshmem program.
-Symmetric data objects are stored on the symmetric heap or in the global/static
-memory section of each \ac{PE}.
+memory regions on different \acp{PE}.
+Figure~\ref{fig:mem_model} shows an example \openshmem
+memory layout, illustrating the location of remotely accessible symmetric
+objects and private data objects.  As shown, symmetric data objects can be
+located either in the symmetric heap or in the global/static memory section of
+each \ac{PE}.
+
+\subsection{Pointers to Symmetric Objects}
+
+Symmetric data objects are referenced in \openshmem operations through the
+local pointer to the desired remotely accessible object.  The address contained
+in this pointer is referred to as a {\em symmetric address}.  Every symmetric
+address is also a {\em local address} that is valid for direct memory access;
+however, not all local addresses are symmetric.  Manipulation of symmetric
+addresses passed to \openshmem routines, including pointer arithmetic;
+structure or union member access; and array indexing are permitted as long as
+the resulting local pointer remains within the same symmetric allocation or
+object.  Symmetric addresses are only valid at the \ac{PE} where they were
+generated; using a symmetric address generated by a different \ac{PE} results
```

Comment:
just a minor suggestion for clarity:

using a symmetric address generated by a different \ac{PE} for direct memory access or as an argument to an \openshmem routine results in undefined behavior.